core/src/main/scala/chisel3/dontTouch.scala

```diff
@@ -46,4 +47,18 @@ object dontTouch {
     data
   }
 
+  /** Mark a module so that its ports will not get optimized away.
+    *
+    * @param module whose ports shouldl not get optimized away
+    */
+  def modulePorts(module: BaseModule): BaseModule = {
+
+    annotate(new ChiselMultiAnnotation {
+      def toFirrtl: Seq[Annotation] = (DataMirror.fullModulePorts(module).map {
+        case (_, data) => DontTouchAnnotation(data.toNamed)
+      })
```

Comment:
`fullModulePorts` is returns every Aggregate and Element so this will result in dontTouching things multiple times. Can we instead maybe just use `.modulePorts` and call `dontTouch.apply` above (which does the recursion and only marks `Elements`)?